capsul/process/process.py

```diff
@@ -736,10 +736,8 @@ def set_parameter(self, name, value):
         value: object (mandatory)
             the trait value we want to set
         """
-        # Detect File and Directory trait types with None value
-        if value is None and is_trait_pathname(self.trait(name)):
-
-            # The None trait value is _Undefined, do the replacement
+        # The None trait value is _Undefined, do the replacement
```

Comment:
Documentation should be updated. I also suspect that None may be a valid value in some cases. But since I have no real example in mind, I suggest to make sure that there is a test with a pipeline that fail if a `None` value is set on a wrong parameter type. This way, if Capsul get modified to make `None` available in some cases, we will be sure that it does not break anything.